src/output/generator.js

```diff
@@ -16,14 +18,44 @@ const PARAMETER_NAME_INDEX = 2;
 const DEFAULT_CSS = path.join(__dirname, '..', 'resources', 'output', 'styles.css');
 const DEFAULT_TEMPLATE = path.join(__dirname, '..', 'resources', 'output', 'templates', 'outputTemplate.html');
 
-const PARAMETER_REGEX = '\\[.+?\\]\\(#[\\w\\d]+\\)';
-const TEST_REGEX = '\\[&?.+?\\]\\(\\?=[\\w\\d]+\\.[\\w\\d]+\\(.*?\\)\\)';
-const ELEMENT_DISCRIMINANT_REGEX = '\\?=';
+// t | test | v | var | variable | i | inv | invoker | m | mod | module
+// [](invoker: PHPInvoker)
+// [](module: LeModule)
+// [valeur](variable: nomDeLaVariable)
+// [valeurAttendue](test: Class.NomDuTest(parametre1, parametre2, parametre3))
+
+const INVOKER_DISCRIMINANTS_REGEX = 'i|inv|invoker';
+
+const MODULE_DISCRIMINANTS_REGEX = 'm|mod|module';
+
+const TEST_DISCRIMINANTS_REGEX = '\\?=|t|test';
+const TEST_ELEMENTS_REGEX = `(?:\\[[&]?(.*?)\\])\\((?:${TEST_DISCRIMINANTS_REGEX})\\:?\\s?(?:(.*?)\\.)?(.*?)\\((.*)\\)\\)`;
+const TEST_PARAMETER_NAME_REGEX = '[\\w\\d]+';
+
+const PARAMETER_DISCRIMINANTS_REGEX = '#|variable|var|v';
+const PARAMETER_ELEMENTS_REGEX = `\\[(.*)?\\]\\((?:${PARAMETER_DISCRIMINANTS_REGEX})\\:?\\s?([\\w\\d]+)\\)`;
+
+const ELEMENT_DISCRIMINANTS_REGEX = `${INVOKER_DISCRIMINANTS_REGEX}|${MODULE_DISCRIMINANTS_REGEX}|${PARAMETER_DISCRIMINANTS_REGEX}|${TEST_DISCRIMINANTS_REGEX}`;
+const ELEMENT_DETECTION_REGEX = `\\[.*?\\]\\((?:${ELEMENT_DISCRIMINANTS_REGEX})\\:?\\s?.*?\\)+`;
+
+const DISCRIMINANT_DETECTION_REGEX = `\\]\\((${ELEMENT_DISCRIMINANTS_REGEX})`;
 
 class HTMLGenerator {
     constructor() {
         this.elementFormattingFunctions = {};
```

Comment:
Would probably be cleaner to do:
```
this.elementFormattingFunctions = {
  i : this._formatInvoker.bind(this),
  inv = this._formatInvoker.bind(this)
}
```
etc

Reply:
Applied requested changes. Left single quotes beause of # symbol and var reserved word.